src/Idris/Completion.hs

```diff
@@ -28,19 +32,21 @@ import qualified Data.Text as T
 import System.Console.ANSI (Color)
 import System.Console.Haskeline
 
+commands :: [String]
 commands = [ n | (names, _, _) <- allHelp ++ extraHelp, n <- names ]
 
 tacticArgs :: [(String, Maybe TacticArg)]
 tacticArgs = [ (name, args) | (names, args, _) <- Idris.Parser.Expr.tactics
                             , name <- names ]
+
+tactics :: [String]
 tactics = map fst tacticArgs
 
 -- | Get the user-visible names from the current interpreter state.
 names :: Idris [String]
-names = do i <- get
-           let ctxt = tt_ctxt i
+names = do ctxt <- tt_ctxt <$> getIState
            return $
-             mapMaybe nameString (allNames $ definitions ctxt) ++
+             mapMaybe nameString (allNames (visibleDefinitions ctxt)) ++
```

Comment:
I think `allNames $ visibleDefinitions ctxt` is more readable (akin to what was before).